### DISCUSSION TITLE
Dark UI

### DISCUSSION THREAD

**Discussion by happysurf** [uBlockOrigin/uBlock-issues#1963] [General] [answered]
Finally the dark UI is here but in the logger, all is still horrible.
Tested on Windows 7 at moment.

**Comment by uBlock-user** [comment]
> all is still horrible.

Elaborate on what is "horrible".

**Comment by gorhill** [comment]
> horrible

How is this constructive? To be constructive you need to be very specific about what needs to be fixed, otherwise we have no idea what you think requires fixing, and no action taken as a result.

**Comment by happysurf** [comment]
Sorry guys you are right.

This screenshot is taken from Firefox 97 final and UBO 1.40.9b11 on Windows 7.

![Dark UI](https://user-images.githubusercontent.com/13637587/152533701-aa0fc1da-5dd5-4ffb-969c-db90a7f59e90.png)

**Reply by gorhill** [reply]
I can reproduce the "Click to preview" issue, I will fix this.

I cannot reproduce the illegible line in the logger, mine is light text on dark background. Maybe you have remnant of customizations for uBO's CSS, you should remove all these to be sure they are not interfering with uBO's official support of dark theme.

**Reply by uBlock-user** [reply]
Web Workers are always shadowed, perhaps shadows won't work with dark theme.

**Reply by gorhill** [reply]
Any link which I can simply navigate to so that I get webworker entries?

**Reply by uBlock-user** [reply]
- Navigate to https://github.com/ with logger open and Dark theme enabled
- Put `0,1` in the filter widget to filter web-worker entries.

**Comment by ghost** [comment]
Blinking white background when dark theme is active.
<details>
<summary>gif</summary>

![1](https://user-images.githubusercontent.com/98751006/152544222-f0e8aee1-5cd8-4a29-bb4a-26c52224ac00.gif)
</details>

<details>

```yaml
uBlock Origin: 1.40.9rc0
Chromium: 98
filterset (summary): 
  network: 96192
  cosmetic: 62224
  scriptlet: 18070
  html: 0
listset (total-discarded, last updated): 
  default: 
    user-filters: 0-0, never
    RUS-0: 41141-9, 7m
    easylist: 62829-20, 7m
    easyprivacy: 26696-113, 7m
    plowe-0: 3677-822, 7m
    ublock-abuse: 77-1, 7m
    ublock-badware: 3831-72, 7m
    ublock-filters: 30183-181, 7m
    ublock-privacy: 194-2, 7m
    ublock-unbreak: 1699-39, 7m
    urlhaus-1: 7611-0, 7m
filterset (user): [empty]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 359 ms (selfie)
  maxAssetCacheWait: 229 ms
popupPanel: 
  blocked: 1
  blockedDetails: 
    github.com: 1
```
</details>

**Reply by uBlock-user** [reply]
@RumpleThump https://github.com/uBlockOrigin/uBlock-issues/issues/1284

**Comment by gorhill** [comment]
@happysurf provide an exact URL which I can use to reproduce exactly what you get in your logger.

---

Ok I forced an update of one of uBO's list, and tabless network requests shows up fine in dark mode, i.e. light text on dark background, so I can't still reproduce that illegible line in your screenshot.

**Comment by u-RraaLL** [comment]
Since we're talking about the logger's dark UI, may I request some changes?

1.  `mediumblue` color for CNAME
<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/92610420/152552374-36a21dfe-e2f4-4b2b-8a5c-90095144d6b8.png)</details>
`mediumblue` is rather hard to read on a dark background. Especially when looking from a distance, suffering from eye-strain or generally for people with slightly worse vision.
`blue` is only _slightly_ better - I'm proposing changing it to something like `deepskyblue` or `cyan`, if we need to stay with blue variants.

Changing the color to one of these lighter hues will also kinda fix the issue with shadowed web workers as mentioned in @happysurf 's comment and @uBlock-user 's reply.

<details><summary>Screenshot for comparison</summary>

![image](https://user-images.githubusercontent.com/92610420/152558514-d3308ac9-dba9-41e2-9e43-b8db63c01f21.png)</details>


2. Remaining elements with white backgrounds
<details><summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/92610420/152559301-50cb687f-93bb-48d8-a1c0-e016de7b5900.png)
I know these fields don't _have to_ be darker, considering their size, but some lightgrey  would be better, IMO. (Also for their Dashboard counterparts - filter/search fields + dropdown fields)

![image](https://user-images.githubusercontent.com/92610420/152559525-f20fdb45-5a60-4005-bd1c-f76c168c026a.png)


![image](https://user-images.githubusercontent.com/92610420/152559945-b7fcad2c-baa7-45ff-9d37-584e31e1428e.png)

![image](https://user-images.githubusercontent.com/92610420/152559680-4d619291-2112-4fdd-acdb-3d6d588efba3.png)
</details>

Could we get these fixed, please?

**Reply by u-RraaLL** [reply]
Reading #1958 just now also made me check this:

![image](https://user-images.githubusercontent.com/92610420/152564405-06eb77bd-408d-4852-88b8-f53ece9027ed.png)

**Reply by ghost** [reply]
CNAME is `#93a6ff` in popup.

**Reply by u-RraaLL** [reply]
> CNAME is #93a6ff in popup.

Yeah, I guess using `#93a6ff`/`var(--popup-cell-cname-ink)` will also work.
Point is, something lighter.

**Reply by gorhill** [reply]
Cname in logger is just something I overlooked, I didn't choose this specifically, the proper one is the one already defined for the popup panel, so I will change this.

For the form widget, I prefer to let the browser or OS deal with these, I would expect it would do so when dark mode is set browser- or system-wide.

**Reply by u-RraaLL** [reply]
I have both system-wide and browse-wide dark modes and the dropdown entries remain dark, it's just the initial fields that show up white in uBO.

![image](https://user-images.githubusercontent.com/92610420/152577215-297ea907-e7b6-48a6-98e8-d39f9791458d.png)

![image](https://user-images.githubusercontent.com/92610420/152577269-4be27fb9-ce00-419e-971d-b112580add69.png)

**Reply by gorhill** [reply]
On my side I have these fields in a light color, including the drop down -- uBO does not change the background color on these, I decided to refrain from doing so a long time ago because of bad side effect in the appearance of built-in widgets. The filter expression overlay is a custom div though, so it will be affected by default background color.

**Comment by happysurf** [comment]
The link for try the logger is https://pixcut.wondershare.com/account.html
I have updated UBO to version 1.40.9rc0 and disabled HA in Firefox, but both the problem are still here.
No additional CSS codes for UBO.

**Reply by gorhill** [reply]
Ok, so these were cname-related entries, fixed with <https://github.com/gorhill/uBlock/commit/35490439b9c35084fb11192724bda8be52612038> as per above feedback.

**Comment by ghost** [comment]
These dark CNAME lines are from closed tabs. Open `github.blog` -> close the tab.

![obraz](https://user-images.githubusercontent.com/886325/152570180-190b6fe7-eadf-44d4-9c64-4f4ee41d127c.png)

**Comment by gorhill** [comment, marked as answer]
> may I request some changes?

Just to be clear, in the [main commit for dark mode](https://github.com/gorhill/uBlock/commit/ecb73d2ff526b48679b8e62ee4ff6bf963fc0fdb), I pointed out that I don't expect it to be complete everywhere, there is a lot of corners in uBO, and that I will fix as per feedback -- this is what we are doing here. So if you see a bad color choice, it's not because I thought this was a good idea, it's just something which I overlooked -- there were a huge amount of CSS-related changes and at some point I had to commit what I had without waiting to have 100% coverage by trying to find everything on my own.

**Reply by uBlock-user** [reply]
All form widgets in all uBO panes are white on dark themes or when set to auto(browser theme set to dark.).

**Reply by gorhill** [reply]
As said, I won't touch form widgets, this is opening a can of worms -- the appearance of these is controlled by the browser. If you expect them to be modified by dark theme in the browser, you will have to open an issue about this with the browser devs.

**Comment by u-RraaLL** [comment]
There's an issue with the selection color in CodeMirror (does picker use CodeMirror too?) if the window becomes inactive. E.g. when snapping two windows to screen edges to test filters on two different browser profiles.

![image](https://user-images.githubusercontent.com/92610420/152590629-c6637fb7-f113-44df-b45f-523b6011b1c9.png)

![image](https://user-images.githubusercontent.com/92610420/152590743-2ab9e982-fb07-4390-8244-f6641390a2b8.png)

![image](https://user-images.githubusercontent.com/92610420/152590979-ec3c1864-d6ad-41c8-a57a-4fd930a14781.png)

**Reply by gorhill** [reply]
I'm going to have to research to find out if it's possible.

**Reply by u-RraaLL** [reply]
Slight correction. The color change also happens in the same browser window when shifting focus to another item (e.g. clicking outside the textarea in "My Filters" or when trying to drag the Picker box whilst its contents are selected).

**Reply by gorhill** [reply]
Should be improved with <https://github.com/gorhill/uBlock/commit/ed25ed6a6ca72f7f8247063e31459bd31d9d74c1>.

**Reply by u-RraaLL** [reply]
Thanks. I'll try it out when the next rc goes live.

**Reply by ghost** [reply]
![obraz](https://user-images.githubusercontent.com/886325/152618201-613a3be2-1cf9-4b40-b8b5-cd049ece5f7e.png)
![obraz](https://user-images.githubusercontent.com/886325/152618275-b1486f15-4a33-44ef-a4e8-921745cd2432.png)
![obraz](https://user-images.githubusercontent.com/886325/152618316-59105ce8-6995-4ca7-b325-db8f30e5b008.png)

**Comment by ghost** [comment]
Dark theme is not applied after session restore. Both in dashboard and logger window.

Firefox 98.0a1, 1.40.9rc1, system and browser set to default theme (light).

**Reply by gorhill** [reply]
You mean when uBO's own pages are already opened as part of the session? If so, this is <https://github.com/uBlockOrigin/uBlock-issues/issues/1669>.

**Comment by Acursen** [comment]
The Click2Load text is not visible.

1. Add ||youtube.com/embed/$3p,frame,redirect=click2load.html
2. Visit [this link](https://www.nvidia.com/en-us/geforce/news/god-of-war-pc-dlss-reflex-out-now/) 
3. Click on the YouTube video

Result:
![JcvT2lL9uK](https://user-images.githubusercontent.com/99033943/152653686-8f98a1b0-b21c-4362-87b0-696750f8661e.png)

**Reply by gorhill** [reply]
Should be fixed with <https://github.com/gorhill/uBlock/commit/e44768f5d37a6b4d99191fd9395446bedc0ba344>.

**Comment by u-RraaLL** [comment]
A minor cosmetic issue - any chances for a slightly higher contrast between the background and the font in Picker's counter? Be it either a darker background or a darker (black?) font.

I know these haven't changed since the Light theme, but I think it's just a tad more difficult for the eye to adjust to white on light grey from the darker colors everywhere else than it is on the light theme. Though, I also did have trouble (on occasion) reading the numbers on there in the past (due to eye strain or wherever).

**Reply by gorhill** [reply]
Fixed with <https://github.com/gorhill/uBlock/commit/ed1b5e20531bc0c3dd421e9a14773acf2c70bca7>.

**Comment by u-RraaLL** [comment]
Just thinking out loud here, but do you think using `opacity` for `.synopsis` in Settings was the best idea? It's not that bad on the Dark theme, but on Light, it kinda reads as "unimportant". At least to me. And considering one of the synopses is for the new "Advanced" section, that seems rather counterproductive...

Perhaps using `font-style: italic` instead of changing opacity would be better? (on both themes)

Also, not entirely related, but I was wondering if using 4 color variables for `.info` icons isn't a bit too much (and with opacity in the synopsis it looks like there's a 2nd shade of grey too)?
AFAIK, `var(--info2-ink)` is only used in the Trusted pane now + is less attention grabbing than `var(--info3-ink)`, which seems to be only in the Advanced section synopsis. And I'm just wondering if the 3 different colored info icons `.important`, `.very-important` and `.mustread` do really need to be different if they're all on different tabs?
Personally, I'd probably stick with `var(--info3-ink)` / `.very-important` for them all.

**Reply by gorhill** [reply]
I use a lot [this guide](https://material.io/design/color/text-legibility.html) to make decisions, and this is the case here. It says medium emphasis should use text at opacity 60, and this is what I did. I checked the resulting color versus the background, and I get a AA+ contrast ratio, which means it passes the contrast test. I prefer to stick to the material design guide to make this sort of decision, otherwise every decision become subjective and as such candidate for endless debate because everybody has their own opinion when subjectivity is involved.

**Reply by u-RraaLL** [reply]
I see. I just figured I'd ask since the opacity is a new addition. Thanks for clarifying.

**Comment by happysurf** [comment]
The shadows problems remain on 1.40.9rc1.

![Untitled-2](https://user-images.githubusercontent.com/13637587/152671215-a04a5ecc-a6e1-4afb-afbf-92036d8d911c.png)

**Reply by gorhill** [reply]
I consider the shadow issue to be a long standing issue, I found it was problematic before as well, and I will need to research something which work better than using a text shadow regardless of theme. The context information is what helps distinguish tabless requests, their root context is always `0,...`.

**Comment by ghcsd** [comment]
<details>
<summary>Screenshot</summary>
<img src=https://user-images.githubusercontent.com/32384764/152686706-6ddee75c-5765-4922-9b98-7dda89fc5420.gif>
</details>

Adding `color-scheme: dark` can improve the color contrast of input elements in dark theme. 
https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme

**Reply by gorhill** [reply]
Thanks for the info, I wasn't awere of `color-scheme`. However in the documentation, for the `normal` (default) value it says:

> Indicates that the element isn't aware of any color schemes, and so should be rendered using the browser's default color scheme.

The browser is rendering these form widgets, the question is how come the browser does not make use of the dark theme for these? Surely the browser is better placed to know whether it should apply dark theme to these widget when `perfers-color-scheme` is set to `dark`.

**Reply by gorhill** [reply]
I am going to refrain using `color-scheme` for now, this needs more testing. For example, the element picker overlay become all opaque when I use `color-theme: dark` -- and I don't want to delay further the next stable release. I could probably fix this by overriding locally with `color-scheme: normal` but this quirk just tells me this would need more testing, so best to wait for next dev cycle.

**Reply by ghost** [reply]
> The browser is rendering these form widgets, the question is how come the browser does not make use of the dark theme for these? Surely the browser is better placed to know whether it should apply dark theme to these widget when `perfers-color-scheme` is set to `dark`.

I think this option is a hint for the browser, it says that **the page** supports dark/light schemes. If the page does not support them (`normal`) then the browser will not try to adjust the widgets.

What wonders me is that from the "Compatibility" section, it looks like the only supported value is `color-scheme: light dark;`???

**Reply by gorhill** [reply]
Supported value is `dark`, `light`, `light dark` -- what is not supported is the modifier `only`.

**Reply by Nojuuu** [reply]
```css
.dark input, .dark select, .dark textarea {
	color-scheme: dark;
}
```

Wouldn't that work?

**Reply by gorhill** [reply]
Thanks for reminding me, I was planning to use `color-scheme` in the current dev cycle but forgot about this.

**Reply by krystian3w** [reply]
IMO no works good #2062

**Comment by ghajini** [comment]
Theme set is "auto"

There's `create` button which was yellow when creating the filters,but now it is grey

<details><summary>Screenshot</summary>

![Screenshot_20220211-204457-670](https://user-images.githubusercontent.com/20338483/153617513-baae68e7-fa8e-4e36-a83d-1edf186bd0be.png)
</details>

is it by design now? if yes then doesn't look good

**Reply by gorhill** [reply]
Fixed with <https://github.com/gorhill/uBlock/commit/9a5acbbfcd3d400da8da449b8f5205870cd4fcc4>.

**Comment by Acursen** [comment]
The icon at the bottom of "my rules" is hard to see in dark mode. But to be fair, it's hard to see in light mode as well.

![FL2RvbP4EZ](https://user-images.githubusercontent.com/99033943/153656682-03a61708-7789-48e8-a190-1aa12b34c6ba.png)

**Reply by gorhill** [reply]
Fixed with <https://github.com/gorhill/uBlock/commit/3e2f7793738c00ed58e86493a65c708906ed4eef>.

**Comment by Nojuuu** [comment]
Could `--sf-keyword-ink` be changed so that it stands out a bit more? At the moment it feels like it blend in too much with the others.

**Reply by gorhill** [reply]
Suggest a replacement color?

**Reply by Nojuuu** [reply]
If you look edit history of this comment you can see that I originally had suggestion, but I edited it out because I thought that maybe it was too aggressive color. But something like that where it clearly stands out that there's asterisks in the url, for example.

**Reply by gorhill** [reply]
What is the hex color code in your screenshot?

**Reply by Nojuuu** [reply]
`#ef0bef`

**Reply by gorhill** [reply]
Your hue is ~308, the original is 280 -- so for the next dev build I will nudge the hue to 290 and wait for feedback.

**Comment by u-RraaLL** [comment]
Logger's `#netFilteringDialog > .headers > .tools > span:hover` is not theme specific, serving a light background on the dark theme.

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/25fe95eb895496fc284095b66d4a3b7eccc8acf0

**Comment by u-RraaLL** [comment]
In the dark theme, the noop rules in the popup panel have reversed colors - inherited ones are dark gray.

https://www.reddit.com/r/uBlockOrigin/comments/te10qc/question_about_noop_colors_light_grey_and_dark/

**Reply by gorhill** [reply]
The inherited ones are closer to the color of the ones without rules, so in the light theme they are lighter gray, in the dark theme they are darker gray. Same applies to the block/allow rules, the lightness of the color on inherited rules is between the lightness of the color of the rule and the default lightness of the background.

**Comment by u-RraaLL** [comment]
A very minor issue. In the logger, when on the "All" tab, the message about filter lists being reloaded is nearly invisible with its dark blue font color on a dark grey background.

---
https://github.com/gorhill/uBlock/commit/ec8292ef92b462cf1a2837fd5168f5dff2570a45
